File: Baekjoon/Problem_1010/Problem_1010.cpp
#include <iostream>

using namespace std;

unsigned long long Factorial(const int n)
{
    unsigned long long output = 1;
    for(int i = n; i > 0; i--) output *= i;
    return output;
}


unsigned long long Permutation(const int n, const int r)
{
    unsigned long long output = 1;
    for(int i = n; i > n - r; i--) output *= i;
    return output;
}

unsigned long long Combination(const int n, const int r)
{
    unsigned long long p = Permutation(n, r);
    unsigned long long f = Factorial(r);
    unsigned long long output = p / f;
    return output;
}

int main(void)
{
    int Testcase; cin >> Testcase;

    for(int i = 0; i < Testcase; i++)
    {
        int N, M; cin >> N >> M;
        if(N > M - N) N = M - N;

        cout << Combination(M, N) << endl;
    }
}

File: Baekjoon/Problem_1000~1999/Problem_1010/Problem_1010.py
def Pectorial(n):
    output = 1
    for i in range(1, n+1): output = output * i
    return output

def Permutation(n, r):
    output = 1
    for i in range(n - r, n + 1): output = output * i
    return output

def Combination(n, r):
    return Permutation(n, r) / Pectorial(r)

def main():
    Testcase = int(input())

    for i in range(Testcase):
        N = int(input())
        M = int(input())

        print(Combination(M, N))

main()